scripts/check_for_package_main.ts

```diff
@@ -0,0 +1,22 @@
+import { glob } from 'glob';
+import { readPackageJson } from './components/package-json/package_json.js';
+
+const packagePaths = await glob('./packages/*');
+
+const errors: string[] = [];
+for (const packagePath of packagePaths) {
+  const { main, name, isPrivate } = await readPackageJson(packagePath);
+  if (isPrivate || name === '@aws-amplify/integration-tests') {
+    continue;
+  }
```

Comment:
```suggestion
  const { main, name, private } = await readPackageJson(packagePath);
  if (private) {
    continue;
  }
```

We can just skip private packages. Their correctness is not that relevant. There seems to be a bug in this logic though. See below.

See https://github.com/aws-amplify/amplify-backend/blob/fe6c23d543b80cff33de9ee8896e05b30afdb4e9/scripts/components/package-json/package_json.ts#L7 .

and

https://github.com/aws-amplify/amplify-backend/blob/fe6c23d543b80cff33de9ee8896e05b30afdb4e9/packages/integration-tests/package.json#L3